File: sandbox-runner/patterns/event-accumulator/typescript/activities.ts
import type { OrderItem } from "./shared";

export async function processItems(orderId: string, items: OrderItem[]): Promise<string> {
  const names = items.map((i) => i.name).join(", ");
  return `Order ${orderId} fulfilled with ${items.length} item(s): ${names}`;
}

File: sandbox-runner/patterns/event-accumulator/python/activities.py
from temporalio import activity

from shared import OrderItem


@activity.defn
async def process_items(order_id: str, items: list[OrderItem]) -> str:
    names = ", ".join(i.name for i in items)
    return f"Order {order_id} fulfilled with {len(items)} item(s): {names}"
